core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponentSpec.scala

```diff
@@ -1350,6 +1350,175 @@ class CompactEntityComponentSpec extends TestDriverComponentWithFlatSpecAndMatch
+
+    val newTargetEntity1 = targetEntity1.copy(entityType = newTargetType)
+    val newTargetEntity2 = targetEntity2.copy(entityType = newTargetType)
+    
+    // Verify entity types were updated in the ENTITY table
+    val newTargets = runAndWait(q.listEntities(wsid, newTargetType))
+    newTargets.map(_.toEntity) should contain theSameElementsAs Seq(newTargetEntity1, newTargetEntity2)
+    
+    // Verify from_entity_type was updated in ENTITY_REFS table
+    runAndWait(q.getReferencesFrom(wsid, sourceEntity1.toPointer)) should contain theSameElementsAs Seq(newTargetEntity1.toPointer, newTargetEntity2.toPointer)
+    runAndWait(q.getReferencesFrom(wsid, sourceEntity2.toPointer)) should contain theSameElementsAs Seq(newTargetEntity2.toPointer)
+    runAndWait(q.getReferencesFrom(wsid, sourceEntity3.toPointer)) should contain theSameElementsAs Seq(newTargetEntity1.toPointer)
+    
+    // Verify old references no longer exist
+    runAndWait(q.getReferencesTo(wsid, Seq(targetEntity1.toPointer))) shouldBe empty
+    runAndWait(q.getReferencesTo(wsid, Seq(targetEntity2.toPointer))) shouldBe empty
+    
+    // Verify to_entity_type was updated in ENTITY_REFS table
+    runAndWait(q.getReferencesTo(wsid, Seq(newTargetEntity1.toPointer))) should contain theSameElementsAs Seq(sourceEntity1.toPointer, sourceEntity3.toPointer)
+    runAndWait(q.getReferencesTo(wsid, Seq(newTargetEntity2.toPointer))) should contain theSameElementsAs Seq(sourceEntity1.toPointer, sourceEntity2.toPointer)
+    
+    // Verify references in entity attributes are updated
+    val updatedSourceEntity1 = runAndWait(q.getEntity(wsid, sourceType, sourceEntity1.name)).get.toEntity
+    updatedSourceEntity1.attributes(AttributeName.withDefaultNS("refList")).asInstanceOf[AttributeEntityReferenceList].list should contain theSameElementsAs Seq(
+      newTargetEntity1.toReference,
+      newTargetEntity2.toReference
+    )
+    val updatedSourceEntity2 = runAndWait(q.getEntity(wsid, sourceType, sourceEntity2.name)).get.toEntity
+    updatedSourceEntity2.attributes(AttributeName.withDefaultNS("ref")) shouldBe newTargetEntity2.toReference
+    
+    val updatedSourceEntity3 = runAndWait(q.getEntity(wsid, sourceType, sourceEntity3.name)).get.toEntity
+    updatedSourceEntity3.attributes(AttributeName.withDefaultNS("refSingletonList")).asInstanceOf[AttributeEntityReferenceList].list should contain theSameElementsAs Seq(
+      newTargetEntity1.toReference
+    )
+    // Verify entity keys table is updated
+    import driver.api._
+
+    val entityKeysQuery1 = sql"""select entity_type from ENTITY_KEYS where workspace_id = $wsid and entity_type = $targetType""".as[String]
+    runAndWait(entityKeysQuery1).size shouldBe 0
+    val entityKeysQuery2 = sql"""select entity_type from ENTITY_KEYS where workspace_id = $wsid and entity_type = $newTargetType""".as[String]
+    runAndWait(entityKeysQuery2).size shouldBe 2
```

Comment:
I like this because it's readable, but FYI if you wanted you could also use `slickDataSource.dataAccess.compactEntityKeysSlickQuery` to generate Slick queries against the ENTITY_KEYS table